strawberry/pagination/__init__.py

```diff
@@ -0,0 +1,21 @@
+from .fields import ConnectionExtension, connection
```

Comment:
Pagination and everything inside it is composed of `Connection` code moved from the `relay` package. No code here is new 😊

Reply:
mmh, I wonder about this, Connection is a relay specific term, no? I wonder if moving it outside will bring confusion 🤔

Reply:
I think relay coined that term, but today it is a general way of doing pagination in GraphQL. An example is this: https://graphql.org/learn/pagination

I have absolutely nothing against keeping it in relay, and calling it "relay style pagination" (as we were discussing in private =P), I'm only afraid of still making people confused, thinking this can only be used with [relay.dev](https://relay.dev/)

Let me ping @strawberry-graphql/core here to get everyone's opinions :)

Reply:
@DoctorJohn my idea in this PR was for (1), but if we go for (2), which is what @patrick91 was suggesting, we are going to limit it from being used with non `Node` objects, which is not something that the connection actually requires (but I got your point)

What do you think about scheduling a discussion on this on discord? @patrick91 is out right now but we can do that once he gets back :)

Reply:
for what is worth, I'm ok with dropping the requirement on Node, but still keep it as relay pagination 😊